docs/canvas/canvas-expression-lifecycle.asciidoc

```diff
@@ -0,0 +1,241 @@
+
+
+[[canvas-function-can-take-arguments]]
+=== Functions Can Take Arguments
+
+Let’s look at another expression, which uses the same <<demodata_fn>> function but this time produces a pie chart instead.
+
+image::images/canvas-functions-can-take-arguments-pie-chart.png[Pie Chart]
+[source,text]
+----
+filters
+| demodata
+| pointseries color="state" size="max(price)"
+| pie
+| render
+----
+
+This one starts out just like before, using the <<filters_fn>> and <<demodata_fn>> functions to produce a filtered set of random data. This time, however, the output becomes the context for <<pointseries_fn>>, which is a way to "aggregate" or "roll up" your data, similar to how Elasticsearch works, but more generalized. In this case, the data is split up using the `color` and `size` dimensions, using arguments on the <<pointseries_fn>> function. Each unique value in the "state" column will have an associated "size" value, which in this case, will be the maximum value of the "price" column.
+
+If the expression stopped there, it would produce a `pointseries` data type as the output of this expression. But instead of looking at the raw values, the result is piped to <<pie_fn>>, which will produce an output that will render a pie visualization. And just like before, this is piped to <<render_fn>>, which is useful for its arguments.
+
+The end result is a simple pie chart that uses the default color palette. But <<pie_fn>> can take some additional arguments that control how it gets rendered. For example, you can provide a `hole` argument to turn your pie chart into a donut chart, one that has a "hole" in the middle of it, but changing the expression to this:
+
+
+image::images/canvas-functions-can-take-arguments-donut-chart.png[Donut Chart]
+[source,text]
+----
+filters
+| demodata
+| pointseries color="state" size="max(price)"
+| pie hole=50
+| render
+----
+
+
+[[canvas-aliases-and-unnamed-arguments]]
+=== Aliases and Unnamed Arguments
+
+Argument definitions have one canonical name, which is always provided in the underlying code, but when used in an expression they often times include aliases that make them easier or faster to type, usually both. 
+
+Take, as one example, the <<mapColumn_fn>> function, which has 2 arguments: `expression`, the expression to use to produce a calculated value, and `name`, the name of column to place the resulting data in. The `expression` argument includes some aliases, namely `exp`, `fn`, and `function`. That means that you can use any of those four options to provide that argument’s value. So in your expression, if you use `mapColumn name=newColumn fn={string example}`, this produces the same output as if you wrote the more verbose `mapColumn name=newColumn expression={string example}`.
```

Comment:
How about:

For example, the <<mapColumn_fn>> function has 2 arguments:
* `expression`- Produces a calculated value
* `name` - The name of the column to place the resulting data in